packages/core/echo/indexing/src/types.ts

```diff
@@ -19,6 +19,8 @@ export type IndexQuery = {
    */
   typename?: string | null;
 
+  // TODO(mykola): Extract to QueryPlanner in future
+  or?: IndexQuery[];
```

Comment:
I think it would be better if this type wasn't hierarchical -- consider merging this clause with the `typename` fieldsto to represent a request for any typeneme from the list.

Reply:
Then it would not support `Filter.or(Filter.not(<something>), <something>)`